File: BOJ 2/10996.cpp
#include <cstdio>
#include <algorithm>
using namespace std;
char board[210][105];
int n;
int main(){
	scanf("%d", &n);
	if (n == 1) printf("*\n");
	else {
		for (int i = 1; i <=205; i++) {
			for (int j = 1; j <=103; j++) {
				board[i][j] = ' ';
			}
		}
		int i = 0;
		for (int j = 1; j <= n; j++){
			bool check = false;
			if (j % 2 == 1) check = false;
			else check = true;
			if (!check) i = 1;
			else i = 2;
			for (;i <= 2 * n; i += 2){
				board[i][j] = '*';
			}
		}
		for (int i = 1; i <= 2 * n; i++) {
			for (int j = 1; j <= n; j++) {
				printf("%c", board[i][j]);
			}
			printf("\n");
		}
	}
	return 0;
}

File: BOJ(Java 공부용)/10996.java
import java.io.*;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        if (n == 1) System.out.println('*');
        else {
            char[][] board = new char[2 * n + 1][n + 1];
            int i = 0;
            for (int j = 1; j <= n; j++) {
                boolean check = false;
                if (j % 2 == 1) check = false;
                else check = true;
                if (!check) i = 1;
                else i = 2;
                for (; i <= 2 * n; i += 2) {
                    board[i][j] = '*';
                }
            }
            for (int j = 1; j <= 2 * n; j++) {
                for (int k = 1; k <= n; k++) {
                    if(board[j][k]=='\0') bw.write(" ");
                    else bw.write(board[j][k]);
                }
                bw.write("\n");
            }
            bw.flush();
            bw.close();
        }
    }
}